### PR TITLE
Make Mute also a player trait

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary

SUMMARY: Features "Making Mute a player trait"

#### Purpose of change

In Magiclysm I created an aura that changes the players form to that of a werewolf.  As a con I wanted the player to be unable to communicate with NPCs for the time of the change.  I gave the aura the Mute trait.  However, mute only impacts NPC's with the trait.

#### Describe the solution

This PR seeks to make mute player impacting.  I've done very little in C++ before so I hope this is a good step in the right direction.

#### Describe alternatives you've considered

Learning nothing new ever again. Begging a dev to do this.  

#### Testing

TBD need to setup a compiling environment.

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -4883,6 +4886,11 @@ bool Character::is_deaf() const
              && in_sleep_state() );
 }
 
+bool Character::is_mute() const
+{
+    return get_effect_int( effect_mute ) > 2 || worn_with_flag( flag_MUTE ) ||
+           has_trait( trait_MUTE ) );
+
```
[the remark this answers is not in the document]
Comment (Maleclypse) [reply src/character.cpp]: I think I've got this the way you want it now. Once I can get astyle working and some other problems resolved, but this line is maybe good now.

File: src/npctalk.cpp [hunk 2]
```diff
@@ -874,1701 +881,1713 @@ std::string dialogue::dynamic_line( const talk_topic &the_topic ) const
...
-        return beta->evaluation_by( *alpha );
-    } else if( topic == "TALK_LOOK_AT" ) {
-        return "&" + beta->short_description();
-    } else if( topic == "TALK_OPINION" ) {
-        return "&" + beta->opinion_text();
-    } else if( topic == "TALK_MIND_CONTROL" ) {
-        if( beta->enslave_mind() ) {
-            return _( "YES, MASTER!" );
+            if( alpha->is_mute() ) {
+                return _( "&You yell, but can't hear form words." );
```
Comment (anothersimulacrum) [inline src/npctalk.cpp:885]:
```suggestion
                return _( "&You yell, but can't form words." );
```

File: src/npctalk.cpp [hunk 3]
```diff
@@ -874,1701 +881,1713 @@ std::string dialogue::dynamic_line( const talk_topic &the_topic ) const
...
+            return beta->evaluation_by( *alpha );
+        } else if( topic == "TALK_LOOK_AT" ) {
+            return "&" + beta->short_description();
+        } else if( topic == "TALK_OPINION" ) {
+            return "&" + beta->opinion_text();
+        } else if( topic == "TALK_MIND_CONTROL" ) {
+            if( beta->enslave_mind() ) {
+                return _( "YES, MASTER!" );
+            }
         }
```
Comment (anothersimulacrum) [inline src/npctalk.cpp:899]:
```suggestion
        }
    }
```

File: src/npctalk.cpp [hunk 4]
```diff
@@ -874,8 +881,11 @@ std::string dialogue::dynamic_line( const talk_topic &the_topic ) const
         alpha->shout();
         if( alpha->is_deaf() ) {
             return _( "&You yell, but can't hear yourself." );
-        } else {
-            return _( "&You yell." );
+            if( alpha->is_mute() ) {
```
Comment (ZhilkinSerg) [inline src/npctalk.cpp:884]: This code is dead, because prior statement is `return`.

Comment (Maleclypse) [reply src/npctalk.cpp:884]: Thank you!  Did you fix this already as when I open npctalk.cpp in atom it looks like your changes?

Comment (ZhilkinSerg) [reply src/npctalk.cpp:884]: No, I did not fix anything. It is not my suggested code - just a place with error.

The suggestion to fix would be to add `        } else {` before `if( alpha->is_mute() ) {`

**Pull request merged**